test/WebJobs.Script.Tests/FileMonitoringServiceTests.cs

```diff
@@ -56,6 +57,53 @@ public void InitializesGetDirectorySnapshot()
             }
         }
 
+        [Fact]
+        public static async Task HostRestarts_OnWatchFilesChange()
```

Comment:
Please also update the existing Configure_FileWatching test to also test this new property.